File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.js
function solution(n, costs) {
    let answer = 0;
    const parent = new Array(n).fill().map((_, i) => i);
    
    const getParent = (child) => {
        if (parent[child] === child) return child;
        return parent[child] = getParent(parent[child]);
    }
    
    costs.sort((a, b) => a[2] - b[2]);
    
    costs.forEach(([a, b, cost]) => {
        const parent_a = getParent(a);
        const parent_b = getParent(b);
        
        if (parent_a !== parent_b) {
            answer += cost;
            parent[parent_b] = parent_a;
        }
    });
    
    return answer;
}

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.py
def solution(n, costs):
    answer = 0
    parent = list(range(n))
    
    def get_parent(child):
        if parent[child] == child:
            return child
        parent[child] = get_parent(parent[child])
        return parent[child]
    
    costs.sort(key=lambda x: x[2])
    
    for a, b, cost in costs:
        parent_a = get_parent(a)
        parent_b = get_parent(b)
        
        if parent_a != parent_b:
            answer += cost
            parent[parent_b] = parent_a
            
    return answer
